### PR TITLE
Use new seed_stack puppet module

### PR DESCRIPTION
**Pull request by JayH5**
praekeltfoundation/seed_stack

~~Have to do a dependencies hack for now...~~

### CODE REVIEW

File: puppet/.librarian/puppet/config [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+LIBRARIAN_PUPPET_DESTRUCTIVE: 'false'
```
Comment (jerith) [inline puppet/.librarian/puppet/config:2]: What does `LIBRARIAN_PUPPET_DESTRUCTIVE: 'false'` mean?

Comment (JayH5) [inline puppet/.librarian/puppet/config:2]: I don't know but it's the default :p

/me clones the repo to grep for the answer :sweat_smile:

Comment (JayH5) [inline puppet/.librarian/puppet/config:2]: With destructive mode true, librarian-puppet will delete the entire `modules` directory before each install.

See [here](https://github.com/rodjek/librarian-puppet#rsync-option) for more info about how installs to the directory usually work.

**Comment by jerith** [comment]
My fresh `vagrant up` fails because `puppet/modules` doesn't exist.

File: puppet/.librarian/puppet/config (hunk 1 above)
Comment (jerith) [inline puppet/.librarian/puppet/config:2]: Thanks.

**Comment by JayH5** [comment]
Aww.. will fix

**Comment by jerith** [comment]
:+1: -- the build failures are unrelated and I can confirm that the VM provisions correctly on my machine.

**Pull request merged**